geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -884,7 +886,9 @@ public boolean initialImagePut(final Object key, final long lastModified, Object
                         }
                       }
                       if (newValue == Token.TOMBSTONE) {
-                        owner.updateSizeOnRemove(key, oldSize);
+                        if (!oldIsTombstone) {
```

Comment:
seems like this should care about old being "isRemoved()" instead of only "isTombstone()".